app/__init__.py

```diff
@@ -140,7 +141,10 @@ def process_spec(spec, process_unindexed=False):
             if field_filter == "object":
                 system_profile_spec_processed[field]["children"] = process_spec(props["properties"], True)
 
-    return system_profile_spec_processed
+        if props.get("x-indexed", False):
```

Comment:
Unless I'm misunderstanding, shouldn't it be like this?
```suggestion
        if not props.get("x-indexed", True):
```
We want the default value to be True, and we only want to add the field to the list if the value is explicitly False

Reply:
Good catch, made the change, thanks!